subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -32,14 +32,17 @@ LOG_MODULE_REGISTER(net_wifi_shell, LOG_LEVEL_INF);
 #ifdef CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ENTERPRISE
 static const char ca_cert_test[] = {
 	#include <wifi_enterprise_test_certs/ca.pem.inc>
+	'\0'
```

Comment:
Hi @krish2718 
I also face the same issue with this point. Just want to share the info from my side.
A. I generated PEM with RSA 3072 by openssl. Console output error info :'mbedtls_x509_crt_parse: X509 - The CRT/CRL/CSR format is invalid, e.g. different type expected -0x2180'.
B. I switched RSA 3072 to RSA 1027, this error info disappeared .

By the way, add info about server PEM. Did you test server cert and server key? It seems cert and key mismatched.
From my side console output : 'PK - Given private key password does not allow for correct decryption'

Reply:
Yes, we mandatorily need to add null-terminator for the cert to be parsed as PEM.

> By the way, add info about server PEM. Did you test server cert and server key? It seems cert and key mismatched.
From my side console output : 'PK - Given private key password does not allow for correct decryption'

I did and it worked for me, you can also verify using OpenSSL.

```
openssl rsa -in samples/net/wifi/test_certs/client-key.pem -check
Enter pass phrase for samples/net/wifi/test_certs/client-key.pem:
RSA key ok
```